paimon-core/src/main/java/org/apache/paimon/io/StatsCollectingSingleFileWriter.java

```diff
@@ -43,7 +43,7 @@
 public abstract class StatsCollectingSingleFileWriter<T, R> extends SingleFileWriter<T, R> {
 
     @Nullable private final SimpleStatsExtractor simpleStatsExtractor;
-    @Nullable private SimpleStatsCollector simpleStatsCollector = null;
```

Comment:
Can you keep it nullable, and just create a field `isStatsDisabled` from 
```
Arrays.stream(statsCollectors)
                        .allMatch(p -> p instanceof NoneSimpleColStatsCollector)
```

Reply:
Fixed